envoy-control-core/src/main/resources/lua/ingress_handler.lua

```diff
@@ -0,0 +1,34 @@
+
+function envoy_on_request(handle)
+    local path = handle:headers():get(":path")
+    local method = handle:headers():get(":method")
+    local service_name = handle:headers():get("x-service-name")
+    local xff_header = handle:headers():get("x-forwarded-for")
+    local metadata = handle:streamInfo():dynamicMetadata()
+    metadata:set("envoy.filters.http.lua", "request.info.path", path)
+    metadata:set("envoy.filters.http.lua", "request.info.method", method)
+    metadata:set("envoy.filters.http.lua", "request.info.service_name", service_name)
+    metadata:set("envoy.filters.http.lua", "request.info.xff_header", xff_header)
+end
+
+function envoy_on_response(handle)
+    local rbacMetadata = handle:streamInfo():dynamicMetadata():get("envoy.filters.http.rbac")
+
+    if rbacMetadata == nil or rbacMetadata["shadow_engine_result"] ~= "denied" then
+        return
+    end
+
+    local lua_metadata = handle:streamInfo():dynamicMetadata():get("envoy.filters.http.lua") or {}
+    local service_name = lua_metadata["request.info.service_name"] or ""
+    local path = lua_metadata["request.info.path"]
+    path = path and "'"..path.."'" or ""
+    local protocol = handle:connection():ssl() == nil and "http" or "https"
```

Comment:
`and` with `or` 😕 , what does this do?

Reply:
```
Another useful idiom is (a and b) or c (or simply a and b or c, because and has a higher precedence than or), which is equivalent to the C expression

    a ? b : c
``` source: https://www.lua.org/pil/3.3.html